testing/src/main/java/org/oppia/android/testing/profile/ProfileTestHelper.kt

```diff
@@ -92,4 +102,17 @@ class ProfileTestHelper @Inject constructor(
     testCoroutineDispatchers.runCurrent()
     return result
   }
+
+  /** ObserveForever is needed in order to complete the Action which returns the LiveData.
+   * Any Action which responds with the LiveData must need to observe it, to update the data
+   * on screen while running espresso test cases.
+   * */
+  fun waitForOperationToComplete(data: LiveData<AsyncResult<Any?>>) {
+    data.observeForever(observer)
+  }
+
+  /** Need to remove the observer as it is observing forever. */
+  fun forceToCompleteOperation(data: LiveData<AsyncResult<Any?>>) {
```

Comment:
I wasn't thinking about using a helper, but since you are I suggest just omitting the remove observer. It's fine for testing purposes just to keep the observer registered.

Reply:
done